### ISSUE TITLE
Different subpaneldefs.php for different Roles

### ISSUE THREAD

**Issue by Abuelodelanada** [salesagility/SuiteCRM#2127]
#### Issue

In each module we have a subpaneldef.php (modules/[MODULE]/metadata/subpaneldef.php) that defines the subpanels SuiteCRM will show in a DetailView.

Many times you need to show different subpanels for different roles.
#### Expected Behavior

Have the hability to show different subpanels for different roles in the same DetailView
#### Actual Behavior

Every role see the same subpanels
#### Possible Fix

We should modify the function open_layout_defs() in include/SubPanle/SubpanelDefinitions.php. This function should evaluate the user' s roles and if there is a file called:
modules/[MODULE]/metadata/subpaneldefs_[ROLENAME].php

But every user may be in several roles, so which subpaneldefs should we use?
A possible solution: Add a boolean field called "primary_role" in the table acl_roles_users (As in securitygroups_users table), so we use the definition for the primary_role

**Comment by samus-aran** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Different subpaneldefs.php for different Roles](https://trello.com/c/5mzVhDRx/77-different-subpaneldefs-php-for-different-roles)

**Comment by samus-aran** [comment]
We have now set up a new home for suggestions at Trello.  All github issues that were labeled 'suggestion' have been moved and will be closed.  Certain ones will be progressed within the new Suggestion Box and may be re-opened.

Announcement of moving Suggestions:
https://suitecrm.com/forum/suggestion-box/13691-moving-suggestions-from-github-to

New SuiteCRM Suggestion Box
https://trello.com/b/Ht7LbMqw/suitecrm-suggestion-box
